clang/lib/Sema/TreeTransform.h

```diff
@@ -7231,12 +7226,16 @@ QualType TreeTransform<Derived>::TransformAttributedType(
   // FIXME: dependent operand expressions?
   if (getDerived().AlwaysRebuild() ||
       modifiedType != oldType->getModifiedType()) {
-    TypeLocBuilder AuxiliaryTLB;
-    AuxiliaryTLB.reserve(TL.getFullDataSize());
-    QualType equivalentType =
-        getDerived().TransformType(AuxiliaryTLB, TL.getEquivalentTypeLoc());
-    if (equivalentType.isNull())
-      return QualType();
+    // Do not transform the equivalent type if it is equal to the modified type.
```

Comment:
It'd be better to explain *why* we don't want to transform (that we don't transform can be gleaned from reading the code).

Reply:
Yeah, that’s a pretty bad comment, I agree...

Reply:
I’ve expanded the comment a bit to make this a bit more informative (hopefully).